ci/lava/tests/psa-arch-tests.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""
+Script to run psa-arch-tests.
+
+The actual tests should already be on the target, this script just runs them
+via mbl-cli and translates the output into something LAVA can understand.
+"""
+import re
+import subprocess
+import sys
+
+SUITES = ["crypto"]
+
+EXPECTED_FAILS = {"239_Testing_crypto_asymmetric_APIs": True}
```

Comment:
As far as I understand, https://github.com/ARMmbed/mbed-crypto/issues/126 is the cause of this test failure and this is a bug. If this is correct, then why shouldn't this test script fail too? Doing this is covering over the fact that it fails. IIUC then I think the behaviour should be changed. The fact if fails should be document e.g. in the commit message.

Reply:
> As far as I understand, ARMmbed/mbed-crypto#126 is the cause of this test failure and this is a bug.

Yep.

> If this is correct, then why shouldn't this test script fail too? Doing this is covering over the fact that it fails. IIUC then I think the behaviour should be changed.

I agree to some extent. I'd prefer us to have a test framework that has a concept of expected failures - then it would be marked as a failure, but when people look at test results after changing code they'll know that the failure wasn't due to their changes.

Lava doesn't have a concept of expected failures though, so we can't do that. If we just marked it as a fail then, in practice, it will cause engineers pain because when they see the "psa-arch-tests" test fail, they'll have to check that it has failed in the same way as we expect.

The hack in this script does provide some of the benefits of a proper expected failure - when the test starts passing (unexpectedly) we'll find out about it (because it will be reported as a failure) and we'll be able to remove it from the expected fails list.

> The fact if fails should be document e.g. in the commit message.

Agree. I'll add a code comment and something in the commit message.

Reply:
I've added a comment in https://github.com/ARMmbed/mbl-core/pull/194/commits/69f7f6b2632ef553a6253640b9c5896fc2c17b7c.

Reply:
@jonathanhaigh-arm @petedyerarm As discussed, we'll manage this as best we can, and unfortunately dealing with the error in this way appears to be the way forward.